pkg/operator/k8sutil/service.go

```diff
@@ -62,6 +62,14 @@ func UpdateService(
 	return clientset.CoreV1().Services(namespace).Update(serviceDefinition)
 }
 
+//DeleteService makes a best effort at deleting a Services, then waits for them to be deleted
+func DeleteService(clientset kubernetes.Interface, namespace, name string) error {
+	_ = func(options *metav1.DeleteOptions) error {
```

Comment:
Why is this in an embedded func? Looks like this function won't ever be called, so the service won't be deleted.